cmake/Platform/Linux/CompilerSettings.cmake

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+if(NOT CMAKE_C_COMPILER AND NOT CMAKE_CXX_COMPILER)
```

Comment:
This should also check against the environment variables:
`CC` and `CXX` whether it is setup or not. CMake will key from them if CMAKE_C_COMPILER and CMAKE_CXX_COMPILER is not set.

https://cmake.org/cmake/help/latest/envvar/CC.html
https://cmake.org/cmake/help/latest/envvar/CXX.html